src/meltano/cli/config.py

```diff
@@ -372,16 +373,30 @@ def set_(
     value: t.Any,
     store: str,
     interactive: bool,
+    from_file: t.TextIO,
 ):
     """Set the configurations' setting `<name>` to `<value>`."""
     if len(setting_name) == 1:
         setting_name = tuple(setting_name[0].split("."))
 
     interaction = InteractiveConfig(ctx=ctx, store=store, extras=False)
+
     if interactive:
         interaction.configure_all()
-    else:
-        interaction.set_value(setting_name=setting_name, value=value, store=store)
+        ctx.exit()
+
+    if from_file:
+        setting_name += (value,)
+        value = from_file.read()
+
+        interaction.set_value(
+            setting_name=setting_name,
+            value=value,
+            store=store,
+        )
+        ctx.exit()
+
+    interaction.set_value(setting_name=setting_name, value=value, store=store)
```

Comment:
No need to exit early here, we can reuse the setting name and value:

```suggestion
    if from_file:
        setting_name += (value,)
        value = from_file.read()

    interaction.set_value(setting_name=setting_name, value=value, store=store)
```

Reply:
@edgarrmondragon there are 3 different flows here, and we are also updating global variable, to prevent side effects shouldn't we explicity exit from each step. wdyt?

Reply:
@edgarrmondragon I think it make sense to reuse these values because, after all, we are updating the global variables.